lib.c

```diff
@@ -184,6 +184,8 @@ int parse_canframe(char *cs, struct canfd_frame *cf) {
 		for (i=0; i<8; i++){
 			if ((tmp = asc2nibble(cs[i])) > 0x0F)
 				return 0;
+
+			/* casting tmp to canid_t to perform left shift of 28 */
```

Comment:
Would it make sense to make tmp a canid_t ?

Reply:
Done